src/invocation.js

```diff
@@ -0,0 +1,80 @@
+import { Address } from './address';
+import { scValToNative } from './scval';
+
+/**
+ * @typedef CreateInvocation
+ * @prop {any}    args          a base64-encoded {@link xdr.CreateContractArgs}
+ * @prop {string} executable    a base64-encoded {@link xdr.ContractExecutable}
+ */
+/**
+ * @typedef ExecuteInvocation
+ * @prop {string} source    the strkey of the contract (C...) being invoked
+ * @prop {string} function  the name of the function being invoked
+ * @prop {any[]}  args      the natively-represented parameters to the function
+ *    invocation (see {@link scValToNative}) for rules on how they're
+ *    represented a JS types
+ */
+/**
+ * @typedef InvocationTree
+ * @prop {'execute' | 'create'} type  the type of invocation occurring, either
+ *    contract creation or host function execution
+ * @prop {CreateInvocation | ExecuteInvocation} args  the parameters to the
+ *    invocation, depending on the type
+ * @prop {InvocationTree[]} invocations   any sub-invocations that (may) occur
+ *    as a result of this invocation (i.e. a tree of call stacks)
+ */
+
+/**
+ * Turns a raw invocation tree into a human-readable format.
+ *
+ * This is designed to make the invocation tree easier to understand in order to
+ * inform users about the side-effects of their contract calls. This will help
+ * make informed decisions about whether or not a particular invocation will
+ * result in what you expect it to.
+ *
+ * @param {xdr.SorobanAuthorizedInvocation} root  the raw XDR of the invocation,
```

Comment:
What do you mean by "raw XDR"? base64 string? js object?

Reply:
the js structure, yeah. string could work too but this is just outlining the api more than a full solution